### PR TITLE
refactor: move `is`, `deepEqual` and `shallowEqual` predicates to `@makeswift/controls`

### PR DESCRIPTION
**Pull request by agurtovoy**
No semantic changes, prep for https://github.com/makeswift/makeswift/pull/814

### CODE REVIEW

File: packages/controls/src/lib/predicates/deepEqual.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { shallowEqual } from './shallowEqual'
+
+const { hasOwnProperty } = Object.prototype
+
+export const deepEqual = (a: unknown, b: unknown): boolean => {
```
Comment (agurtovoy) [inline packages/controls/src/lib/predicates/deepEqual.ts:5]: Switched from default to named export, no other changes.

File: packages/controls/src/lib/predicates/is.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export function is(x: unknown, y: unknown): boolean {
```
Comment (agurtovoy) [inline packages/controls/src/lib/predicates/is.ts:1]: Switched from default to named export, no other changes.

File: packages/controls/src/lib/predicates/shallowEqual.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { is } from './is'
+
+const { hasOwnProperty } = Object.prototype
+
+export const shallowEqual = (a: unknown, b: unknown): boolean => {
```
Comment (agurtovoy) [inline packages/controls/src/lib/predicates/shallowEqual.ts:5]: Switched from default to named export, no other changes.

File: packages/runtime/src/utils/deepEqual.ts [hunk 4]
```diff
@@ -1,27 +1,3 @@
...
-      !deepEqual(a[keysA[i]], b[keysA[i]])
-    )
-      return false
-  }
-
-  return true
-}
+import { deepEqual } from '@makeswift/controls'
 
 export default deepEqual
```
Comment (agurtovoy) [inline packages/runtime/src/utils/deepEqual.ts:3]: Forwarding for backward compatibility.

**Review by arvinpoddar — APPROVED** [review]
Looks good to me

**Review by migueloller — COMMENTED** [review]
I failed again to leave a review before this was merged. Sorry!

I wanted to say that it feels a bit weird that we're exporting utilities from `@makeswift/controls`. It seems that what we probably want is a shared library of utilities. Given this is all in a monorepo, I wonder if we could get away with a package that we don't even publish that's bundled with `@makeswift/controls` and `@makeswift/runtime` at build time using Vite.

**Comment by agurtovoy** [comment]
> It seems that what we probably want is a shared library of utilities.

@migueloller For sure, and I think it should actually be a published package, since we have a number of these copied and pasted into our private monorepo as well. I'm just keeping things simple for now to stay on track.

**Pull request merged**